src/audio-worklet/migration/worklet-recorder/app.js

```diff
@@ -128,57 +142,53 @@ function handleRecording(processorPort, recordingProperties) {
     }
     if (event.data.message === 'SHARE_RECORDING_BUFFER') {
       createRecord(recordingProperties, recordingLength, context.sampleRate,
-          downloadLink, downloadButton, event.data.buffer, player);
+          event.data.buffer);
     }
   };
 
-  recordButton.addEventListener('click', (e) => {
-    isRecording = !isRecording;
+  if (recordingState === RecorderState.UNINITIALIZED) {
+    isRecording = true;
+    processorPort.postMessage({
+      message: 'UPDATE_RECORDING_STATE',
+      setRecording: isRecording,
+    });
+    changeButtonStatus();
+    recordingState = RecorderState.RECORDING;
+  }
 
-    // Inform processor that recording was paused.
+  recordButton.addEventListener('click', (e) => {
+    isRecording = true;
     processorPort.postMessage({
       message: 'UPDATE_RECORDING_STATE',
       setRecording: isRecording,
     });
+    changeButtonStatus();
+  });
 
-    recordText.textContent = isRecording ? 'Stop' : 'Start';
-    downloadButton.disabled = isRecording ? true : false;
+  stopButton.addEventListener('click', (e) => {
+    isRecording = false;
+    processorPort.postMessage({
+      message: 'UPDATE_RECORDING_STATE',
+      setRecording: isRecording,
+    });
+    changeButtonStatus();
   });
 
   return recordingEventCallback;
 }
 
-/**
- * Sets up monitor functionality, allowing user to listen to mic audio live.
- * @param {GainNode} monitorNode Gain node to adjust for monitor gain.
- */
-function setupMonitor(monitorNode) {
-  // Leave audio volume at zero by default.
-  monitorNode.gain.value = 0;
-
-  let isMonitoring = false;
-
-  const monitorButton = document.querySelector('#monitor');
-  const monitorText = monitorButton.querySelector('span');
-
-  monitorButton.addEventListener('click', (event) => {
-    isMonitoring = !isMonitoring;
-    const newVal = isMonitoring ? 1 : 0;
-
-    // Set gain to quickly but smoothly slide to new value.
-    monitorNode.gain.setTargetAtTime(newVal, context.currentTime, 0.01);
-
-    monitorText.textContent = isMonitoring ? 'off' : 'on';
-  });
+function changeButtonStatus() {
+  // Inform processor that recording was paused.
```

Comment:
What's `processor` here?

Reply:
I forget to delete this code comment. Before this version, there is a message post function to inform processor in here and then I remove it from this function. This comment is removed now. Thank for pointing it out!